### ISSUE TITLE
implement more constructions for Hadamard matrices, e.g. for size 116

### ISSUE THREAD

**Issue by dimpase** [sagemath/sage#34690]
<div id="comment:0"></div>

With #32267 done, the next not yet implemented construction is for n=116. This was originally constructed by Williamson, 
cf e.g. Marshall Hall's book https://onlinelibrary.wiley.com/doi/book/10.1002/9781118032862

Implementing Williamson construction would be a natural way here.

Depends on #32267

CC:  @MatteoCati

Component: **combinatorics**

Author: **Matteo Cati**

Branch/Commit: **[`bd025a2`](https://github.com/sagemath/sagetrac-mirror/commit/bd025a2e87389b9b7f5573049ad62cdd087e2bf0)**

Reviewer: **Dima Pasechnik**

_Issue created by migration from https://trac.sagemath.org/ticket/34690_

**Comment by dimpase** [comment]
Description changed:
``````diff
--- 
+++ 
@@ -1,4 +1,4 @@
 With #32267 done, the next not yet implemented construction is for n=116. This was originally constructed by Williamson, 
 cf e.g. Marshall Hall's book https://onlinelibrary.wiley.com/doi/book/10.1002/9781118032862
 
-Implementing Williamson construction would be a natiral way here.
+Implementing Williamson construction would be a natural way here.
``````

**Comment by MatteoCati** [comment]
Branch: **[u/gh-MatteoCati/add_williamson_contruction_hadamard_matrices](https://github.com/sagemath/sagetrac-mirror/tree/u/gh-MatteoCati/add_williamson_contruction_hadamard_matrices)**

**Comment by MatteoCati** [comment]
Author: **Matteo Cati**

**Comment by 7ed8c4ca-6d56-4ae9-953a-41e42b4ed313** [comment]
Commit: **[`30ae555`](https://github.com/sagemath/sagetrac-mirror/commit/30ae5550321f39667ba04de4e718eb5fcd53f553)**

**Comment by 7ed8c4ca-6d56-4ae9-953a-41e42b4ed313** [comment]
<div id="comment:4"></div>

Branch pushed to git repo; I updated commit sha1. New commits:
<table><tr><td><a href="https://github.com/sagemath/sagetrac-mirror/commit/57a2d1dbe644b7ed4e34f3e3944d2e0297908825"><code>57a2d1d</code></a></td><td><code>Add test failing when matrix can be generated by paleyI</code></td></tr><tr><td><a href="https://github.com/sagemath/sagetrac-mirror/commit/822b99a16841fda42d0123b424dae1076f7b2b47"><code>822b99a</code></a></td><td><code>Try recursive method in hadamard_matrix only if it will be successful</code></td></tr><tr><td><a href="https://github.com/sagemath/sagetrac-mirror/commit/c1a371e91e505c1abccdf1bdf7d3f12678ad5efa"><code>c1a371e</code></a></td><td><code>Add tests for matrices created by skew_hadamard_matrix and regular_symmetric_hadamard_matrix_with_constant_diagonal</code></td></tr><tr><td><a href="https://github.com/sagemath/sagetrac-mirror/commit/a411049f1df98af9d26281e2f62e72ac375e49ed"><code>a411049</code></a></td><td><code>Add skew_hadamard_matrix and regular_symmetric_hadamard_matrix_with_constant_diagonal to hadamard_matrix function</code></td></tr><tr><td><a href="https://github.com/sagemath/sagetrac-mirror/commit/86506ceecdec2c6e2b843409625b74aa741bae5b"><code>86506ce</code></a></td><td><code>Fix failed test returning Unknown in hadamard_matrix</code></td></tr><tr><td><a href="https://github.com/sagemath/sagetrac-mirror/commit/ec342042b933ed457cf22246d77cdd7a10f724b2"><code>ec34204</code></a></td><td><code>docstring fixes for Payley constructions</code></td></tr><tr><td><a href="https://github.com/sagemath/sagetrac-mirror/commit/ab88569dbcf99bc03de4274d0099eee3537c5d27"><code>ab88569</code></a></td><td><code>Add williamson type hadamard matrix construction</code></td></tr><tr><td><a href="https://github.com/sagemath/sagetrac-mirror/commit/2b8d2982cc2d48ecc024aae954c64f981d2d905d"><code>2b8d298</code></a></td><td><code>Add contruction of williamson type hadamard matrix for n=116 and n=172</code></td></tr><tr><td><a href="https://github.com/sagemath/sagetrac-mirror/commit/06f3cf2e3d3f305d6cfdd8c7158d218e4d25d345"><code>06f3cf2</code></a></td><td><code>Always check williamson necessary conditions</code></td></tr><tr><td><a href="https://github.com/sagemath/sagetrac-mirror/commit/30ae5550321f39667ba04de4e718eb5fcd53f553"><code>30ae555</code></a></td><td><code>Add williamson method to general hadamard matrix contruction</code></td></tr></table>

**Comment by MatteoCati** [comment]
<div id="comment:5" align="right">comment:5</div>

Added Williamson construction, together with data for creating hadamard matrices of order 116 and 172 using this construction.

**Comment by MatteoCati** [comment]
Dependencies: **#32267**

**Comment by 7ed8c4ca-6d56-4ae9-953a-41e42b4ed313** [comment]
Changed commit from **[`30ae555`](https://github.com/sagemath/sagetrac-mirror/commit/30ae5550321f39667ba04de4e718eb5fcd53f553)** to **[`1519681`](https://github.com/sagemath/sagetrac-mirror/commit/1519681122d3099251060e07ae3aab258986cc42)**

**Comment by 7ed8c4ca-6d56-4ae9-953a-41e42b4ed313** [comment]
<div id="comment:7"></div>

Branch pushed to git repo; I updated commit sha1. New commits:
<table><tr><td><a href="https://github.com/sagemath/sagetrac-mirror/commit/1519681122d3099251060e07ae3aab258986cc42"><code>1519681</code></a></td><td><code>Fix documentation for williamson contruction</code></td></tr></table>

**Comment by dimpase** [comment]
<div id="comment:8" align="right">comment:8</div>

OK, good.

The 1st and 2nd smallest value for which we don't have an implementation. 
are now 156 and 188.

156 was first constructed in https://projecteuclid.org/journals/bulletin-of-the-american-mathematical-society/volume-71/issue-1/A-new-construction-for-Hadamard-matrices/bams/1183526410.full 

236 is the 3rd.
Here seems to be the reference on the 1st construction of Hadamard matrices of order 236: https://www.sciencedirect.com/science/article/pii/0097316574900569

For 188 there is https://arxiv.org/pdf/0704.0640.pdf - but most probably it's not the 1st such a construction.

**Comment by dimpase** [comment]
<div id="comment:9" align="right">comment:9</div>

Do we always mean return a normalised Hadamard matrix?

Note that `hadamard_matrix(116)` fails, as there is a default `check` 

```
    if check:
        assert is_hadamard_matrix(M, normalized=True)
```
which is done with `normalized=True` - which is not the case here.

**Comment by dimpase** [comment]
<div id="comment:11" align="right">comment:11</div>

Please also check whether new matrices at #32267 work with `hadamard_matrix()`, i.e. we had this regression already...

**Comment by 7ed8c4ca-6d56-4ae9-953a-41e42b4ed313** [comment]
Changed commit from **[`1519681`](https://github.com/sagemath/sagetrac-mirror/commit/1519681122d3099251060e07ae3aab258986cc42)** to **[`0e639f4`](https://github.com/sagemath/sagetrac-mirror/commit/0e639f4d003b6fad8ca5f0ceb07d0e39073bbb63)**

**Comment by 7ed8c4ca-6d56-4ae9-953a-41e42b4ed313** [comment]
<div id="comment:12"></div>

Branch pushed to git repo; I updated commit sha1. This was a forced push. New commits:
<table><tr><td><a href="https://github.com/sagemath/sagetrac-mirror/commit/1f2bd704f13f617447d3a2b5dfe6ae849971543b"><code>1f2bd70</code></a></td><td><code>Do not check for normalized matrix in hadmard_matrix</code></td></tr><tr><td><a href="https://github.com/sagemath/sagetrac-mirror/commit/ea6ee675cae1e4c1cccd756d29458b249769ffed"><code>ea6ee67</code></a></td><td><code>Add williamson type hadamard matrix construction</code></td></tr><tr><td><a href="https://github.com/sagemath/sagetrac-mirror/commit/a0288915de70200d1560bfacafcd2b8f6021529e"><code>a028891</code></a></td><td><code>Add contruction of williamson type hadamard matrix for n=116 and n=172</code></td></tr><tr><td><a href="https://github.com/sagemath/sagetrac-mirror/commit/a20dc70bc66a2b9271bc3444b61c2aac8828fe8a"><code>a20dc70</code></a></td><td><code>Always check williamson necessary conditions</code></td></tr><tr><td><a href="https://github.com/sagemath/sagetrac-mirror/commit/e69d10b3139c3f3688d7a7ce0129792225932f1f"><code>e69d10b</code></a></td><td><code>Add williamson method to general hadamard matrix contruction</code></td></tr><tr><td><a href="https://github.com/sagemath/sagetrac-mirror/commit/5f8b7a87cb55fa064471571a9fac028b17a9a0b3"><code>5f8b7a8</code></a></td><td><code>Fix documentation for williamson contruction</code></td></tr><tr><td><a href="https://github.com/sagemath/sagetrac-mirror/commit/581e5564d68c3298e2681b0ce11522663bdaad74"><code>581e556</code></a></td><td><code>Add construction for order  156</code></td></tr><tr><td><a href="https://github.com/sagemath/sagetrac-mirror/commit/b5319f01294c8c26c1e95ffaf33a456af9db8636"><code>b5319f0</code></a></td><td><code>Add documentation for hadamard_matrix_156</code></td></tr><tr><td><a href="https://github.com/sagemath/sagetrac-mirror/commit/0e639f4d003b6fad8ca5f0ceb07d0e39073bbb63"><code>0e639f4</code></a></td><td><code>Add matrix of order 156 to general hadamard matrix construction</code></td></tr></table>

**Comment by MatteoCati** [comment]
<div id="comment:13" align="right">comment:13</div>

Replying to [Dima Pasechnik](#comment%3A11):
> Please also check whether new matrices at #32267 work with `hadamard_matrix()`, i.e. we had this regression already...

This error was already present in #32267, so I fixed it in that ticket and then rebased this one.

**Comment by dimpase** [comment]
<div id="comment:14" align="right">comment:14</div>

rebased over latest beta, slight reference fix

---
Last 10 new commits:
<table><tr><td><a href="https://github.com/sagemath/sagetrac-mirror/commit/74b71f5f8d9b34a35d4e41b85753edd1ec427774"><code>74b71f5</code></a></td><td><code>docstring fixes for Payley constructions</code></td></tr><tr><td><a href="https://github.com/sagemath/sagetrac-mirror/commit/17c5755979a4283eebb6a8678fa352cb527248cb"><code>17c5755</code></a></td><td><code>Do not check for normalized matrix in hadmard_matrix</code></td></tr><tr><td><a href="https://github.com/sagemath/sagetrac-mirror/commit/b8132e715e8a66df5c67422f853074eb0065bd28"><code>b8132e7</code></a></td><td><code>Add williamson type hadamard matrix construction</code></td></tr><tr><td><a href="https://github.com/sagemath/sagetrac-mirror/commit/7ae0d7829aafd77252d755be15316865f87475db"><code>7ae0d78</code></a></td><td><code>Add contruction of williamson type hadamard matrix for n=116 and n=172</code></td></tr><tr><td><a href="https://github.com/sagemath/sagetrac-mirror/commit/07c7803283a899fe34b648bf5175f58b9c3ebaab"><code>07c7803</code></a></td><td><code>Always check williamson necessary conditions</code></td></tr><tr><td><a href="https://github.com/sagemath/sagetrac-mirror/commit/d4f42905cd594b63d85bb9f0dc56272ba5ca16ae"><code>d4f4290</code></a></td><td><code>Add williamson method to general hadamard matrix contruction</code></td></tr><tr><td><a href="https://github.com/sagemath/sagetrac-mirror/commit/0bfcbd9acb7b3ce72ed7e0077341745d539771df"><code>0bfcbd9</code></a></td><td><code>Fix documentation for williamson contruction</code></td></tr><tr><td><a href="https://github.com/sagemath/sagetrac-mirror/commit/644f926fc6b2efeb51d4ae680b624a67e8fd5f0b"><code>644f926</code></a></td><td><code>Add construction for order  156</code></td></tr><tr><td><a href="https://github.com/sagemath/sagetrac-mirror/commit/2e3b39795e0fe95a8554ccd48b4decb27948c64e"><code>2e3b397</code></a></td><td><code>Add documentation for hadamard_matrix_156</code></td></tr><tr><td><a href="https://github.com/sagemath/sagetrac-mirror/commit/9bac7b5c999fb375c67c1c8d1ce45f531fc45266"><code>9bac7b5</code></a></td><td><code>Add matrix of order 156 to general hadamard matrix construction</code></td></tr></table>

**Comment by dimpase** [comment]
Reviewer: **Dima Pasechnik**

**Comment by dimpase** [comment]
Changed branch from **[u/gh-MatteoCati/add_williamson_contruction_hadamard_matrices](https://github.com/sagemath/sagetrac-mirror/tree/u/gh-MatteoCati/add_williamson_contruction_hadamard_matrices)** to **[u/dimpase/combinat/add_williamson_contruction_hadamard_matrices](https://github.com/sagemath/sagetrac-mirror/tree/u/dimpase/combinat/add_williamson_contruction_hadamard_matrices)**

**Comment by dimpase** [comment]
Changed commit from **[`0e639f4`](https://github.com/sagemath/sagetrac-mirror/commit/0e639f4d003b6fad8ca5f0ceb07d0e39073bbb63)** to **[`9bac7b5`](https://github.com/sagemath/sagetrac-mirror/commit/9bac7b5c999fb375c67c1c8d1ce45f531fc45266)**

**Comment by 7ed8c4ca-6d56-4ae9-953a-41e42b4ed313** [comment]
Changed commit from **[`9bac7b5`](https://github.com/sagemath/sagetrac-mirror/commit/9bac7b5c999fb375c67c1c8d1ce45f531fc45266)** to **[`7a9910f`](https://github.com/sagemath/sagetrac-mirror/commit/7a9910fa326890190f8b19177272955dc6cef40b)**

**Comment by 7ed8c4ca-6d56-4ae9-953a-41e42b4ed313** [comment]
<div id="comment:15"></div>

Branch pushed to git repo; I updated commit sha1 and set ticket back to needs_review. New commits:
<table><tr><td><a href="https://github.com/sagemath/sagetrac-mirror/commit/7a9910fa326890190f8b19177272955dc6cef40b"><code>7a9910f</code></a></td><td><code>slight fix of the reference format</code></td></tr></table>

**Comment by vbraun** [comment]
<div id="comment:17" align="right">comment:17</div>

Merge failure on top of:

b8151dc1099 Trac #34681: Error with multiplication of points on elliptic curves over Integers(n)

0fcca83be81 Trac #34651: Add `multivariate_interpolation` for multivariate polynomial rings

7c37d8de415 Trac #34611: fast implementation of exp

0a63c62c2da Trac #34397: pycodestyle cleanup in src/sage/graphs/generic_graph.py (part 9)

1f9b781eee1 Trac #34081: Upgrade scipy to 1.9.x, add meson toolchain

58c9281ec67 Trac #32267: make hadamard_matrix() use all the Hadamard matrices Sage knows

edfa1f93738 Trac #30423: F-Matrix Factory

7aa226ced2f Trac #34699: some details about INPUT and INPUT in the doc

c4079b7488c Trac #34697: minor fixes in doc in pyx files

813072800d9 Trac #34691: companion matrix of constant polynomial has the wrong parent

6b70584ab66 Trac #34638: refresh the file categories/rings.py

f634f6b0195 Trac #34368: implement the F,H,M triangles

859c3516190 Trac #34689: make Compositions() an additive monoid

b5082885f9a Trac #34665: openssl spkg-configure.m4: Also require openssl if curl needs to be built

487f2f9bca8 Trac #34662: sage.combinat.permutation.from_cycles produces wrong result when 'cycles' is a generator

ab0944d9eaf Trac #34636: make sparsity a decision of the user

24d0a8a1219 Trac #34381: Add infinite q-Pochhammer symbol

ef1d3d23588 Trac #34260: Implement northwest diagrams

bff11ac3b18 Trac #33176: Fix a few cython "referenced before assignment" warnings

89af3469590 Trac #32267: make hadamard_matrix() use all the Hadamard matrices Sage knows

c3028e74d80 Updated [SageMath](../wiki/SageMath) version to 9.8.beta3



merge was not clean: conflicts in src/sage/combinat/matrices/hadamard_matrix.py

**Comment by dimpase** [comment]
<div id="comment:18" align="right">comment:18</div>

let's wait for the new beta, then rebase

**Comment by MatteoCati** [comment]
Changed commit from **[`7a9910f`](https://github.com/sagemath/sagetrac-mirror/commit/7a9910fa326890190f8b19177272955dc6cef40b)** to none

**Comment by MatteoCati** [comment]
Changed branch from **[u/dimpase/combinat/add_williamson_contruction_hadamard_matrices](https://github.com/sagemath/sagetrac-mirror/tree/u/dimpase/combinat/add_williamson_contruction_hadamard_matrices)** to **[u/gh-MatteoCati/add_more_hadamard_matrices_constructions](https://github.com/sagemath/sagetrac-mirror/tree/u/gh-MatteoCati/add_more_hadamard_matrices_constructions)**

**Comment by 7ed8c4ca-6d56-4ae9-953a-41e42b4ed313** [comment]
Commit: **[`a40a54a`](https://github.com/sagemath/sagetrac-mirror/commit/a40a54a5fb6c674a8f533d027f7892cbc4b9141f)**

**Comment by 7ed8c4ca-6d56-4ae9-953a-41e42b4ed313** [comment]
<div id="comment:20"></div>

Branch pushed to git repo; I updated commit sha1. Last 10 new commits:
<table><tr><td><a href="https://github.com/sagemath/sagetrac-mirror/commit/2e3b39795e0fe95a8554ccd48b4decb27948c64e"><code>2e3b397</code></a></td><td><code>Add documentation for hadamard_matrix_156</code></td></tr><tr><td><a href="https://github.com/sagemath/sagetrac-mirror/commit/9bac7b5c999fb375c67c1c8d1ce45f531fc45266"><code>9bac7b5</code></a></td><td><code>Add matrix of order 156 to general hadamard matrix construction</code></td></tr><tr><td><a href="https://github.com/sagemath/sagetrac-mirror/commit/7a9910fa326890190f8b19177272955dc6cef40b"><code>7a9910f</code></a></td><td><code>slight fix of the reference format</code></td></tr><tr><td><a href="https://github.com/sagemath/sagetrac-mirror/commit/c6c061a64c6767faf5be0b1083a6809591c0ec4f"><code>c6c061a</code></a></td><td><code>Fix doctests in hadamard_matrix_williamson_type</code></td></tr><tr><td><a href="https://github.com/sagemath/sagetrac-mirror/commit/9bb5efd6580d5462c65665fbb49389c241bfafb9"><code>9bb5efd</code></a></td><td><code>Move williamson quadruples to external function</code></td></tr><tr><td><a href="https://github.com/sagemath/sagetrac-mirror/commit/e0d548775dd0b51a039bb9fce1239d5873dddb98"><code>e0d5487</code></a></td><td><code>Add constructions of 4-symbol delta codes</code></td></tr><tr><td><a href="https://github.com/sagemath/sagetrac-mirror/commit/411c5e0c97fa3d8883d8df0321f2371c96ff9346"><code>411c5e0</code></a></td><td><code>Add construction of Baumert-Hall units</code></td></tr><tr><td><a href="https://github.com/sagemath/sagetrac-mirror/commit/ad22bf494301a5405721b68c54e893c96e22fa3b"><code>ad22bf4</code></a></td><td><code>Add Turyn construction of hadamard matrices</code></td></tr><tr><td><a href="https://github.com/sagemath/sagetrac-mirror/commit/ec95a4a159f0a6739d64898e7df74906d7f0f762"><code>ec95a4a</code></a></td><td><code>Add Turyn construction with available data</code></td></tr><tr><td><a href="https://github.com/sagemath/sagetrac-mirror/commit/a40a54a5fb6c674a8f533d027f7892cbc4b9141f"><code>a40a54a</code></a></td><td><code>Fix baumert-hall units construction and related tests</code></td></tr></table>

**Comment by MatteoCati** [comment]
<div id="comment:21" align="right">comment:21</div>

Added Turyn construction for matrix of order 236.
For order 188 I believe the first construction is the one described in: https://ro.uow.edu.au/cgi/viewcontent.cgi?article=1956&context=infopapers

This can be used to construct a few other matrices with the T-sequences given in http://mathscinet.ru/files/OD4t.pdf

**Comment by 7ed8c4ca-6d56-4ae9-953a-41e42b4ed313** [comment]
<div id="comment:22"></div>

Branch pushed to git repo; I updated commit sha1. New commits:
<table><tr><td><a href="https://github.com/sagemath/sagetrac-mirror/commit/3f7b66cf135b4afdd535f2ae10fac6ef8ef4d6d4"><code>3f7b66c</code></a></td><td><code>Fix documentation</code></td></tr></table>

**Comment by 7ed8c4ca-6d56-4ae9-953a-41e42b4ed313** [comment]
Changed commit from **[`a40a54a`](https://github.com/sagemath/sagetrac-mirror/commit/a40a54a5fb6c674a8f533d027f7892cbc4b9141f)** to **[`3f7b66c`](https://github.com/sagemath/sagetrac-mirror/commit/3f7b66cf135b4afdd535f2ae10fac6ef8ef4d6d4)**

**Comment by 7ed8c4ca-6d56-4ae9-953a-41e42b4ed313** [comment]
Changed commit from **[`3f7b66c`](https://github.com/sagemath/sagetrac-mirror/commit/3f7b66cf135b4afdd535f2ae10fac6ef8ef4d6d4)** to **[`815549c`](https://github.com/sagemath/sagetrac-mirror/commit/815549c58623b6fe8a354801fae34e0f055a9f45)**

**Comment by 7ed8c4ca-6d56-4ae9-953a-41e42b4ed313** [comment]
<div id="comment:23"></div>

Branch pushed to git repo; I updated commit sha1. New commits:
<table><tr><td><a href="https://github.com/sagemath/sagetrac-mirror/commit/c21f6989e9749a41bff47241f2b9500f647ab7fe"><code>c21f698</code></a></td><td><code>Fix format of INPUT blocks in documentation</code></td></tr><tr><td><a href="https://github.com/sagemath/sagetrac-mirror/commit/ee3a8b42ff575364525eb62725a9ca0640b92842"><code>ee3a8b4</code></a></td><td><code>Add file with T-sequences constructions</code></td></tr><tr><td><a href="https://github.com/sagemath/sagetrac-mirror/commit/55fa736e3f9d443782075f5029ca2aa2e29d93cd"><code>55fa736</code></a></td><td><code>Create separate function for Goethals Seidel matrix</code></td></tr><tr><td><a href="https://github.com/sagemath/sagetrac-mirror/commit/fbbf5a39be885cb8d4f6f1ba27dde04cf28736d6"><code>fbbf5a3</code></a></td><td><code>Add Cooper Wallis construction</code></td></tr><tr><td><a href="https://github.com/sagemath/sagetrac-mirror/commit/cac74d6ca29ed374d235dcdf72299a97e6383593"><code>cac74d6</code></a></td><td><code>Add smallcases function for Cooper Wallis construction</code></td></tr><tr><td><a href="https://github.com/sagemath/sagetrac-mirror/commit/e04f2123e61281620d7ca8712cfc20fb7e9eda18"><code>e04f212</code></a></td><td><code>Add Cooper Wallis construction to general hadamard_matrix method</code></td></tr><tr><td><a href="https://github.com/sagemath/sagetrac-mirror/commit/815549c58623b6fe8a354801fae34e0f055a9f45"><code>815549c</code></a></td><td><code>Add T-matrices of order 67 in Cooper Wallis construction</code></td></tr></table>

**Comment by 7ed8c4ca-6d56-4ae9-953a-41e42b4ed313** [comment]
<div id="comment:24"></div>

Branch pushed to git repo; I updated commit sha1. New commits:
<table><tr><td><a href="https://github.com/sagemath/sagetrac-mirror/commit/ae16bc089811ff27274eff4feeb6c75414d89c67"><code>ae16bc0</code></a></td><td><code>Fix T-sequences documentation</code></td></tr></table>

**Comment by 7ed8c4ca-6d56-4ae9-953a-41e42b4ed313** [comment]
Changed commit from **[`815549c`](https://github.com/sagemath/sagetrac-mirror/commit/815549c58623b6fe8a354801fae34e0f055a9f45)** to **[`ae16bc0`](https://github.com/sagemath/sagetrac-mirror/commit/ae16bc089811ff27274eff4feeb6c75414d89c67)**

**Comment by 7ed8c4ca-6d56-4ae9-953a-41e42b4ed313** [comment]
Changed commit from **[`ae16bc0`](https://github.com/sagemath/sagetrac-mirror/commit/ae16bc089811ff27274eff4feeb6c75414d89c67)** to **[`2f5a449`](https://github.com/sagemath/sagetrac-mirror/commit/2f5a4497db381877a094d439dc20f0fae9d89f35)**

**Comment by 7ed8c4ca-6d56-4ae9-953a-41e42b4ed313** [comment]
<div id="comment:25"></div>

Branch pushed to git repo; I updated commit sha1. New commits:
<table><tr><td><a href="https://github.com/sagemath/sagetrac-mirror/commit/e37a0ba21e5435c8c5f6a9e14a3325120f14d060"><code>e37a0ba</code></a></td><td><code>Rename T_sequecences.py to lowercase</code></td></tr><tr><td><a href="https://github.com/sagemath/sagetrac-mirror/commit/2f5a4497db381877a094d439dc20f0fae9d89f35"><code>2f5a449</code></a></td><td><code>Add T-sequences file to manual</code></td></tr></table>

**Comment by dimpase** [comment]
<div id="comment:26" align="right">comment:26</div>

new beta has been released, please rebase

**Comment by MatteoCati** [comment]
Changed commit from **[`2f5a449`](https://github.com/sagemath/sagetrac-mirror/commit/2f5a4497db381877a094d439dc20f0fae9d89f35)** to **[`53c98c8`](https://github.com/sagemath/sagetrac-mirror/commit/53c98c894132a882e3503241c452350b0fee50e3)**

**Comment by MatteoCati** [comment]
Changed branch from **[u/gh-MatteoCati/add_more_hadamard_matrices_constructions](https://github.com/sagemath/sagetrac-mirror/tree/u/gh-MatteoCati/add_more_hadamard_matrices_constructions)** to **[u/gh-MatteoCati/add_hadamard_matrices_constructions](https://github.com/sagemath/sagetrac-mirror/tree/u/gh-MatteoCati/add_hadamard_matrices_constructions)**

**Comment by MatteoCati** [comment]
<div id="comment:27"></div>

Last 10 new commits:
<table><tr><td><a href="https://github.com/sagemath/sagetrac-mirror/commit/991e6eb88a15e005f3dcbedd2ecf638e0fee6e1d"><code>991e6eb</code></a></td><td><code>Add file with T-sequences constructions</code></td></tr><tr><td><a href="https://github.com/sagemath/sagetrac-mirror/commit/bb6587dd02bc840f9d032e16397ec8ffc2a40dae"><code>bb6587d</code></a></td><td><code>Create separate function for Goethals Seidel matrix</code></td></tr><tr><td><a href="https://github.com/sagemath/sagetrac-mirror/commit/ec6384b6a544d91de3042fe20a47683525c1936b"><code>ec6384b</code></a></td><td><code>Add Cooper Wallis construction</code></td></tr><tr><td><a href="https://github.com/sagemath/sagetrac-mirror/commit/1f955fcdc0fd7e72bd4725b13d91804b4ee512d3"><code>1f955fc</code></a></td><td><code>Add smallcases function for Cooper Wallis construction</code></td></tr><tr><td><a href="https://github.com/sagemath/sagetrac-mirror/commit/41a56fc56f87eefa375412707357444bb779a924"><code>41a56fc</code></a></td><td><code>Add Cooper Wallis construction to general hadamard_matrix method</code></td></tr><tr><td><a href="https://github.com/sagemath/sagetrac-mirror/commit/59f103bd88cdc8ffd3a66adeebf0f58dcc7581b5"><code>59f103b</code></a></td><td><code>Add T-matrices of order 67 in Cooper Wallis construction</code></td></tr><tr><td><a href="https://github.com/sagemath/sagetrac-mirror/commit/0bcdf4242948f9197d90421c06935213ac7692d3"><code>0bcdf42</code></a></td><td><code>Fix T-sequences documentation</code></td></tr><tr><td><a href="https://github.com/sagemath/sagetrac-mirror/commit/26440afe56b8cba5c1d8b5456c8806bf7e6ce465"><code>26440af</code></a></td><td><code>Rename T_sequecences.py to lowercase</code></td></tr><tr><td><a href="https://github.com/sagemath/sagetrac-mirror/commit/122a947d2d9282e3dc74b9fc9584622ee4c85033"><code>122a947</code></a></td><td><code>Add T-sequences file to manual</code></td></tr><tr><td><a href="https://github.com/sagemath/sagetrac-mirror/commit/53c98c894132a882e3503241c452350b0fee50e3"><code>53c98c8</code></a></td><td><code>Fix docstrings format</code></td></tr></table>

**Comment by MatteoCati** [comment]
<div id="comment:28" align="right">comment:28</div>

Replying to [Dima Pasechnik](#comment%3A26):
> new beta has been released, please rebase

Done.

**Comment by dimpase** [comment]
<div id="comment:29" align="right">comment:29</div>

still "needs work" ?

**Comment by dimpase** [comment]
<div id="comment:31" align="right">comment:31</div>

In docstrings, please keep Hadamard capitalised.

**Comment by 7ed8c4ca-6d56-4ae9-953a-41e42b4ed313** [comment]
<div id="comment:32"></div>

Branch pushed to git repo; I updated commit sha1. New commits:
<table><tr><td><a href="https://github.com/sagemath/sagetrac-mirror/commit/163d780eb758e40b47624922708e71e94578c247"><code>163d780</code></a></td><td><code>Fix typos in docstrings</code></td></tr></table>

**Comment by 7ed8c4ca-6d56-4ae9-953a-41e42b4ed313** [comment]
Changed commit from **[`53c98c8`](https://github.com/sagemath/sagetrac-mirror/commit/53c98c894132a882e3503241c452350b0fee50e3)** to **[`163d780`](https://github.com/sagemath/sagetrac-mirror/commit/163d780eb758e40b47624922708e71e94578c247)**

**Comment by MatteoCati** [comment]
<div id="comment:33" align="right">comment:33</div>

The next construction that can be implemented is the one described in https://core.ac.uk/download/pdf/82189436.pdf, which will give new Hadamard matrices for many different orders (including 292, which is the smallest value currently missing).

**Comment by dimpase** [comment]
<div id="comment:34" align="right">comment:34</div>

OK, great. For [comment:33](#comment%3A33) constructions, please open a follow-up ticket. 

How far will we be from 668 (the 1st 4k for which existence of a matrix is unknown) then?

**Comment by 7ed8c4ca-6d56-4ae9-953a-41e42b4ed313** [comment]
<div id="comment:35"></div>

Branch pushed to git repo; I updated commit sha1 and set ticket back to needs_review. New commits:
<table><tr><td><a href="https://github.com/sagemath/sagetrac-mirror/commit/ef50ecd594f5d3ca2a17d6f7965c08d27c1cf673"><code>ef50ecd</code></a></td><td><code>Add note for typo in reference for T-sequences construction</code></td></tr></table>

**Comment by 7ed8c4ca-6d56-4ae9-953a-41e42b4ed313** [comment]
Changed commit from **[`163d780`](https://github.com/sagemath/sagetrac-mirror/commit/163d780eb758e40b47624922708e71e94578c247)** to **[`ef50ecd`](https://github.com/sagemath/sagetrac-mirror/commit/ef50ecd594f5d3ca2a17d6f7965c08d27c1cf673)**

**Comment by MatteoCati** [comment]
<div id="comment:36" align="right">comment:36</div>

Replying to [Dima Pasechnik](#comment%3A34):
> How far will we be from 668 (the 1st 4k for which existence of a matrix is unknown) then? 

After implementing the construction in [comment:33](#comment%3A33) the missing orders will be 412, 428, 452, 476, 508, 532, 604, 612, 652 but it should be possible to implement all of them just by adding more data (T-sequence and williamson type matrices) to the constructions that are already present.

**Comment by 7ed8c4ca-6d56-4ae9-953a-41e42b4ed313** [comment]
Changed commit from **[`ef50ecd`](https://github.com/sagemath/sagetrac-mirror/commit/ef50ecd594f5d3ca2a17d6f7965c08d27c1cf673)** to **[`3b5f11c`](https://github.com/sagemath/sagetrac-mirror/commit/3b5f11c7d1cdc9b10b8642693b4ebf470f29b8f2)**

**Comment by 7ed8c4ca-6d56-4ae9-953a-41e42b4ed313** [comment]
<div id="comment:37"></div>

Branch pushed to git repo; I updated commit sha1. This was a forced push. New commits:
<table><tr><td><a href="https://github.com/sagemath/sagetrac-mirror/commit/3b5f11c7d1cdc9b10b8642693b4ebf470f29b8f2"><code>3b5f11c</code></a></td><td><code>Add note for typo in reference for T-sequences construction</code></td></tr></table>

**Comment by MatteoCati** [comment]
<div id="comment:38" align="right">comment:38</div>

I've created the next ticket: #34807

**Comment by dimpase** [comment]
<div id="comment:39" align="right">comment:39</div>

Could you replace `sets` in `...-sequences are sets of four (-1, 0, 1) sequences` (2 times) in `combinat/t_sequences.py` with `tuples` ? (At least in the 2nd case it's obvious that each of the 4 -1,0,1-sequences plays a distinct role, so they can't be just be swapped arbitrarily.

**Comment by 7ed8c4ca-6d56-4ae9-953a-41e42b4ed313** [comment]
Changed commit from **[`3b5f11c`](https://github.com/sagemath/sagetrac-mirror/commit/3b5f11c7d1cdc9b10b8642693b4ebf470f29b8f2)** to **[`b936e18`](https://github.com/sagemath/sagetrac-mirror/commit/b936e188336b054ac3a3eb545cfc9c3caa21744a)**

**Comment by 7ed8c4ca-6d56-4ae9-953a-41e42b4ed313** [comment]
<div id="comment:40"></div>

Branch pushed to git repo; I updated commit sha1. New commits:
<table><tr><td><a href="https://github.com/sagemath/sagetrac-mirror/commit/b936e188336b054ac3a3eb545cfc9c3caa21744a"><code>b936e18</code></a></td><td><code>Fix documentation of T sequences</code></td></tr></table>

**Comment by dimpase** [comment]
Changed branch from **[u/gh-MatteoCati/add_hadamard_matrices_constructions](https://github.com/sagemath/sagetrac-mirror/tree/u/gh-MatteoCati/add_hadamard_matrices_constructions)** to **[u/dimpase/add_hadamard_matrices_constructions](https://github.com/sagemath/sagetrac-mirror/tree/u/dimpase/add_hadamard_matrices_constructions)**

**Comment by dimpase** [comment]
<div id="comment:41"></div>

New commits:
<table><tr><td><a href="https://github.com/sagemath/sagetrac-mirror/commit/021fbe5fa512b8eec5698eb7b1922b8581477fba"><code>021fbe5</code></a></td><td><code>adding ref to the typo in old paper, and cosmetic changes</code></td></tr><tr><td><a href="https://github.com/sagemath/sagetrac-mirror/commit/bd025a2e87389b9b7f5573049ad62cdd087e2bf0"><code>bd025a2</code></a></td><td><code>refs to biblio file, trimmed trailing whitespaces, docstring fix</code></td></tr></table>

**Comment by dimpase** [comment]
Changed commit from **[`b936e18`](https://github.com/sagemath/sagetrac-mirror/commit/b936e188336b054ac3a3eb545cfc9c3caa21744a)** to **[`bd025a2`](https://github.com/sagemath/sagetrac-mirror/commit/bd025a2e87389b9b7f5573049ad62cdd087e2bf0)**

**Comment by dimpase** [comment]
<div id="comment:42" align="right">comment:42</div>

looks good.

**Comment by vbraun** [comment]
Changed branch from **[u/dimpase/add_hadamard_matrices_constructions](https://github.com/sagemath/sagetrac-mirror/tree/u/dimpase/add_hadamard_matrices_constructions)** to **[`bd025a2`](https://github.com/sagemath/sagetrac-mirror/commit/bd025a2e87389b9b7f5573049ad62cdd087e2bf0)**
